src/real/real_sigmaScript.sml

```diff
@@ -26,126 +28,90 @@ val POP_ORW = POP_ASSUM (fn thm => ONCE_REWRITE_TAC [thm]);
     Where f's range is the real numbers and P is finite.
    ---------------------------------------------------------------------- *)
 
-val REAL_SUM_IMAGE_DEF = new_definition(
-  "REAL_SUM_IMAGE_DEF",
-  ``REAL_SUM_IMAGE f s = ITSET (\e acc. f e + acc) s (0:real)``);
+(* The old definition (only used by REAL_SUM_IMAGE_THM):
+
+Definition REAL_SUM_IMAGE_DEF :
+    REAL_SUM_IMAGE f s = ITSET (\e acc. f e + acc) s (0:real)
+End
+
+   The new definition (based iterateTheory):
+ *)
+Definition REAL_SUM_IMAGE_DEF :
+    REAL_SUM_IMAGE f s = iterate$Sum s f
```

Comment:
Both definitions are essentially the same. `REAL_SUM_IMAGE` is defined using `ITSET` to sum over uncountable as well as uncountable sets. On the other hand, `iterate.sum` is defined using `iterate (+)` which is potentially boils down to the same definition in `SUM_CLAUSES`as `REAL_SUM_IMAGE_THM`.


 I think instead of redefining `REAL_SUM_IMAGE` on `sum`. You can simply prove an equivalency relationship between `iterate.sum` and `REAL_SUM_IMAGE` and switch to `sum` at your ease. The use of `REAL_SUM_IMAGE` is prevalent in most of the subsequent theories, which might break with your change.